BringYourOwnBoard.md

```diff
@@ -0,0 +1,157 @@
+3. The image build system MUST bake a hardware compatibility file into the image. The Device Update proof-of-concept creates this within the [recipes-support/adu-swupdate-hw-compat.bb](./recipes-support/adu-swupdate-hw-compat/adu-swupdate-hw-compat.bb) but you can choose to use whatever file or versioning method you would like. 
+4. If you are using the [meta-azure-device-update repository](https://github.com/Azure/meta-azure-device-update) meta layer to bake the image in you must set the `ADU_SOFTWARE_VERSION` to the version of Device Update you are using within your build system. 
+
+These are general guidances on larger scale items that need to be taken care of for the agent to build properly for your device. The following sections will focus on what the Device Update `meta-raspberrypi-adu` layer is attempting to accomplish in each layer and what considerations you may want to make. If you read the above items and are unsure of how to proceed please reach out to the Device Update team or talk to your BSP provider. Likely an answer to your question will require information from both. 
+
+# On Patches In this Repo
+
+There are a lot of patches within this repository. This is a normal process of developing meta-layers for Yocto. They're intended only for the `meta-raspberrypi` layer. You don't need to think about them, however you might find that you do need to update and or edit your existing BSP with some patches of your own. You can safely disregard them in your port. 
+
+# Recipes
+
+## recipes-bsp 
+
+### What does it do? 
+
+Within our project `recipes-bsp` is used for appending instructions to the bootloader that allow SwUpdate to set which partition to boot into after completing an image update (eg if we've completed an update while running on `/dev/mmcblk0p1` then when the device restarts we boot into `/dev/mmcblk0p2` which contains the new version of the image). For our purposes we can use [rpi-u-boot-scr.bbappend](./recipes-bsp/rpi-u-boot-scr/rpi-u-boot-scr.bbappend) to add the [boot.cmd.in](./recipes-bsp/rpi-u-boot-scr/files/boot.cmd.in) file to the Raspberry Pi 4's boot process which instructs the device to do the A/B switching. 
+
+Within this layer we edit u-boot to use our own firmware environment configuration which sets aside a part of memory for us to store the variable that we use to indicate to the bootloader which partition to boot into. 
+
+There is also a patch for a GCC issue some RaspberryPi's run into. The reader can safely disregard this and any other patch. They are BSP specific and should not be required for your build. 
+
+### What do I need to think about? 
+
+You must talk to your BSP provider and determine what bootloader they provide as apart of the BSP. Likely you will need to create a more durable u-boot script that handles detecting if an update occurred, determines what partition to load into, and provide the logic to do that switching. 
+
+For instance you can look at this snippet: 
```

Comment:
it should link to the script at the bottom; otherwise, these could get out-of-sync